cpp/src/arrow/filesystem/s3fs.cc

```diff
@@ -2670,21 +2868,13 @@ struct AwsInstance : public ::arrow::internal::Executor::Resource {
   std::atomic<bool> is_finalized_;
 };
 
-std::shared_ptr<AwsInstance> CreateAwsInstance() {
-  auto instance = std::make_shared<AwsInstance>();
-  // Don't let S3 be shutdown until all Arrow threads are done using it
-  arrow::internal::GetCpuThreadPool()->KeepAlive(instance);
-  io::internal::GetIOThreadPool()->KeepAlive(instance);
-  return instance;
-}
-
-AwsInstance& GetAwsInstance() {
-  static auto instance = CreateAwsInstance();
-  return *instance;
+AwsInstance* GetAwsInstance() {
+  static auto instance = std::make_shared<AwsInstance>();
```

Comment:
We may want to use `std::make_unique()` instead because this smart pointer isn't shared.